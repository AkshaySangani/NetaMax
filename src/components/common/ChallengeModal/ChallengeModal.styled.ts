import { ModalBody } from "@chakra-ui/react";
import styled from "styled-components";

export const GradientBody = styled(ModalBody)`
  background: radial-gradient(#1379e1, #2253c9);
  border-radius: 12px;
  padding: 35px 8px 0 8px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  display: flex;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), inset -22px -10px 71px rgba(116, 0, 0, 0.25);
  padding: 35px 20px 25px 20px;
`;
