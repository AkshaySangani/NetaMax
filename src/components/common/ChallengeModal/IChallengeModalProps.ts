export interface IChallengeModalProps {
  /**
   * indicates if the modal is open.
   * @type {boolean}
   */
  isOpen: boolean;

  /**
   * function when clicking close modal icon.
   * @type {() => void}
   */
  onClose: () => void;

  /**
   * function when clicking the modal button.
   * @type {() => void}
   */
  onAccept: () => void;
}
