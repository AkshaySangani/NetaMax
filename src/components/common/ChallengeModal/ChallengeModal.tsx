import React, { ReactElement } from "react";

import ChallengeComplete from "styled/icons/ChallengeComplete";
import { screenSizes } from "styled/screen";

import {
  useMediaQuery,
  Button,
  Modal,
  ModalCloseButton,
  ModalContent,
  ModalOverlay,
  Text,
} from "@chakra-ui/react";

import { GradientBody } from "./ChallengeModal.styled";
import { IChallengeModalProps } from "./IChallengeModalProps";

/**
 *  Modal component that shows when completing the available challenges.
 * @param {IChallengeModalProps} props the component props
 * @returns {ReactElement} The modal component
 */
export const ChallengeModal = ({
  isOpen,
  onClose,
  onAccept,
}: IChallengeModalProps): ReactElement => {
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      size={isDesktop ? "4xl" : "sm"}
      scrollBehavior="inside"
      preserveScrollBarGap
      isCentered
    >
      <ModalOverlay />
      <ModalContent
        borderRadius="13px"
        minH="345px"
        marginLeft="20px"
        marginRight="20px"
        data-testid="challenge-modal-content"
      >
        <ModalCloseButton fontSize="18px" color="#FFF" data-testid="challenge-modal-close-button" />
        <GradientBody w="100%" maxHeight="550px" backgroundColor="red">
          <ChallengeComplete width="125px" height="77px" />
          <Text fontWeight="700" color="#FFF" fontSize="24px" marginBottom="15px">
            ¡Felicitaciones!
          </Text>
          <Text fontWeight="700" color="#FFF" fontSize="18px" textAlign="center">
            Llegaste a la meta, tendrás 10 moNETAs para usar en tu próxima compra
          </Text>
          <Button
            borderRadius="16px"
            onClick={onAccept}
            h="56px"
            w="290px"
            minH="56px"
            backgroundColor="#FFF"
            color="#3870FF"
            _disabled={{ bgColor: "netaGray.300", cursor: "no-drop", color: "#FFF" }}
            _hover={{ bgColor: "none" }}
            data-testid="challenge-modal-button"
            marginTop="40px"
          >
            ¡Entendido!
          </Button>
        </GradientBody>
      </ModalContent>
    </Modal>
  );
};
