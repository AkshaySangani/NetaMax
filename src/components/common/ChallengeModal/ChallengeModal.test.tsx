import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { ChallengeModal } from "./ChallengeModal";
import { IChallengeModalProps } from "./IChallengeModalProps";

describe("Render ChallengeModal Component", () => {
  const props: IChallengeModalProps = {
    onAccept: jest.fn(),
    onClose: jest.fn(),
    isOpen: true,
  };

  test("ChallengeModal shows up and render properly", () => {
    const { container } = render(<ChallengeModal {...props} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("challenge-modal-content")).toBeDefined();
  });

  afterEach(() => {
    cleanup();
  });
});

describe("Test ChallengeModal component functions", () => {
  const props: IChallengeModalProps = {
    onAccept: jest.fn(),
    onClose: jest.fn(),
    isOpen: true,
  };

  it("execute onClose when clicking the close button", () => {
    render(<ChallengeModal {...props} />);
    fireEvent.click(screen.getByTestId("challenge-modal-close-button"));
    expect(props.onClose).toBeCalled();
  });

  it("execute onAccept when clicking the modal button", () => {
    render(<ChallengeModal {...props} />);
    fireEvent.click(screen.getByTestId("challenge-modal-button"));
    expect(props.onAccept).toBeCalled();
  });
});
