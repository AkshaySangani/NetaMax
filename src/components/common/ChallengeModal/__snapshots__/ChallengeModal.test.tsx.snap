// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ChallengeModal Component ChallengeModal shows up and render properly 1`] = `<div />`;
